### PR TITLE
fix(droid): fileopenpicker ui thread

### PR DESCRIPTION
**Pull request by ajpinedam**
GitHub Issue (If applicable): closes #

## PR Type

What kind of change does this PR introduce?

- Bugfix

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

**Comment by MartinZikmund** [comment]
Is this the case on WinUI? I would expect the picker requires the caller to execute from the UI thread 🤔

**Review by agneszitte — APPROVED** [review]
@ajpinedam I imagine you will add some tests for these related changes?

File: src/Uno.UWP/Storage/Pickers/FileOpenPicker.Android.cs [hunk 1]
```diff
@@ -42,15 +52,30 @@ internal static bool TryHandleIntent(Intent intent, Result resultCode)
...
 		{
-			var files = await PickFilesAsync(false, token);
-			return files.Count == 0 ? null : files[0];
+			var tcs = new TaskCompletionSource<StorageFile?>();
+			NativeDispatcher.Main.Enqueue(async () =>
+			{
+				var files = await PickFilesAsync(false, token);
+				var file = files.Count > 0 ? files[0] : null;
+
+				tcs.TrySetResult(file);
```
Comment (jeromelaban) [inline src/Uno.UWP/Storage/Pickers/FileOpenPicker.Android.cs:63]: We should try/catch and call `TrySetException` as well, same for the others.

Comment (ajpinedam) [reply src/Uno.UWP/Storage/Pickers/FileOpenPicker.Android.cs]: Added!

**Comment by ajpinedam** [comment]
> @ajpinedam I imagine you will add some tests for these related changes?

No able to test. This just added non-UI thread support to the API. We did similar work for iOS; this PR is just looking for equal functionality.

**Comment by ajpinedam** [comment]
@jeromelaban  this PR is ready to be reviewed!

**Pull request merged**